src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            result.append(name).append(" - ");
+            int sum = 0;
+            for (String info : data) {
+                if (info.contains(name) && dateValid(dateFrom, info.substring(0, 10), dateTo)) {
+                    sum += Integer.parseInt(info.split(" ")[2])
+                            * Integer.parseInt(info.split(" ")[3]);
+                }
```

Comment:
we don't want to split our string many times (so we can save our splitted result to a variable and reuse it)
[Any magic numbers should be constants](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#any-magic-numbers-should-be-constants)
we can introduce constants like `HOURS_WORKED_INDEX` or something else